Integrations/python/deephaven/Types.py

```diff
@@ -0,0 +1,229 @@
+#
+# Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+#
+"""
+Basic Deephaven table related data types
+"""
+
+import jpy
+import wrapt
+import sys
+import collections
+
+# None until the first _defineSymbols() call
+_table_tools_ = None
+_col_def_ = None
+_python_tools_ = None
+_jclazz_type_ = None
+_qst_col_header_ = None
+_qst_column_ = None
+_qst_newtable_ = None
+_table_ = None
+
+def _defineSymbols():
+    if not jpy.has_jvm():
+        raise SystemError("No java functionality can be used until the JVM has been initialized through the jpy module")
+
+    global _table_tools_, _col_def_, _python_tools_, _jclazz_type_, _qst_col_header_, _qst_column_, _qst_newtable_, _table_
+    if _table_tools_ is None:
+        # This will raise an exception if the desired object is not the classpath
+        _table_tools_ = jpy.get_type("io.deephaven.db.tables.utils.TableTools")
+        _col_def_ = jpy.get_type("io.deephaven.db.tables.ColumnDefinition")
+        _python_tools_ = jpy.get_type("io.deephaven.integrations.python.PythonTools")
+        _jclazz_type_ = jpy.get_type("java.lang.Class")
+        _qst_col_header_ = jpy.get_type("io.deephaven.qst.column.header.ColumnHeader")
+        _qst_column_ =  jpy.get_type("io.deephaven.qst.column.Column")
+        _qst_newtable_ =  jpy.get_type("io.deephaven.qst.table.NewTable")
+        _table_ = jpy.get_type("io.deephaven.db.tables.Table")
+
+# every method that depends on symbols defined via _defineSymbols() should be decorated with @_passThrough
+@wrapt.decorator
+def _passThrough(wrapped, instance, args, kwargs):
+    """
+    For decoration of module methods, to define necessary symbols at runtime
+
+    :param wrapped: the method to be decorated
+    :param instance: the object to which the wrapped function was bound when it was called
+    :param args: the argument list for `wrapped`
+    :param kwargs: the keyword argument dictionary for `wrapped`
+    :return: the decorated version of the method
+    """
+
+    _defineSymbols()
+    return wrapped(*args, **kwargs)
+
+try:
+    _defineSymbols()
+except Exception as e:
+    pass
+
+@_passThrough
+def asType(type_str):
+    return _table_tools_.typeFromName(type_str)
+
+bool_ = asType('java.lang.Boolean')
```

Comment:
So, numba and pandas represent bool_ as a byte; and I don't think they have the concept of a "null" boolean. Does this make our booleans incompatible?

Reply:
Without providing some kind of conversion, probably.  Do we need a dh.bool_from_byte method?